src/macros.jl

```diff
@@ -706,55 +712,48 @@ function build_constraint(
 end
 
 function build_constraint(
-    _error::Function,
-    v::AbstractJuMPScalar,
+    ::Function,
+    v::Union{AbstractJuMPScalar},
     set::MOI.AbstractScalarSet,
 )
     return ScalarConstraint(v, set)
 end
+function _clear_constant!(expr::Union{GenericAffExpr,GenericQuadExpr})
+    offset = constant(expr)
+    add_to_expression!(expr, -offset)
+    return expr, offset
+end
+function _clear_constant!(α::Number)
+    return zero(α), α
+end
 function build_constraint(
-    _error::Function,
-    expr::Union{GenericAffExpr,GenericQuadExpr},
+    ::Function,
+    expr::Union{Number,GenericAffExpr,GenericQuadExpr},
     set::MOI.AbstractScalarSet,
 )
     if MOI.Utilities.supports_shift_constant(typeof(set))
-        offset = constant(expr)
-        add_to_expression!(expr, -offset)
+        expr, offset = _clear_constant!(expr)
         new_set = MOI.Utilities.shift_constant(set, -offset)
         return ScalarConstraint(expr, new_set)
     else
         return ScalarConstraint(expr, set)
     end
 end
-function build_constraint(
-    _error::Function,
-    α::Number,
-    set::MOI.AbstractScalarSet,
-)
-    return build_constraint(_error, convert(AffExpr, α), set)
-end
 function build_constraint(
     _error::Function,
     ::_MA.Zero,
     set::MOI.AbstractScalarSet,
 )
-    return build_constraint(_error, zero(AffExpr), set)
+    return build_constraint(_error, false, set)
```

Comment:
Here's the culprit! I don't know why this method exists. But it seems horrific.

Reply:
I guess it get's called when you have an empty sum or something like that. It will be converted by `model_convert` before getting passed to `JuMP.add_constraint`